### PR TITLE
Regarding issue #759 and 758

### LINKED ISSUE #758: Adding the ability for Privacy Policy Slide
Privacy Policy slide requires hyperlinks enabled and probably a modification to the next button making it say I agree.

### LINKED ISSUE #759: ISlidePolicy not working for me
The app is running without errors and everything is working except the policy respect part since the slide movement doesn't get disabled and regardless of anything the user is able to continue to the next page.

Here is my fragment code

```
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.Toast;

import com.github.appintro.ISlidePolicy;

public class privacy_policy extends Fragment implements ISlidePolicy {
    public privacy_policy() {
    // Removing or adding this part made no difference but I added it anyway since google added 
     it when I created a blank fragment.
    }
    private CheckBox checkBox;
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
// Tried getView both on Create and on Create View didn't work
        checkBox = getView().findViewById(R.id.agree);
    }
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View test = inflater.inflate(R.layout.privacy_policy_layout, container, false);
// I also tried doing test.findViewById and it did not work
        checkBox = test.findViewById(R.id.agree);
        return test;
    }
    @Override
    public boolean isPolicyRespected() {
     return checkBox.isChecked();

    }
    @Override
    public void onUserIllegallyRequestedNextPage() {
        Toast.makeText(getContext(), "Please click on the checkbox to continue", Toast.LENGTH_SHORT).show();
    }
}
```

Here is my XML code 

I only added the part that makes it refer to the java class so I do not annoy you with big chunks of XML code

```
<?xml version="1.0" encoding="utf-8"?>
<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    tools:context=".privacy_policy">
```

and to connect the slide to the intro I added this line to my AppIntro activity

`addSlide(AppIntroCustomLayoutSlide.newInstance(R.layout.privacy_policy_layout));`

So is this a bug or is there something I am doing wrong?

I am using the AppIntro GitHub repository version. I downloaded it and added it to my android project as a module and enabled kotlin for sure.
Since the 5.1.0 doesn't have the custom slide feature.

Thank you for your time and your awesome project. I really appreciate it.

### PR DESCRIPTION
**Pull request by Supermiro19**
This pull request is regarding issue #759 so other people do not encounter my problem when using this awesome library.

Closes #759 , Closes #758

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -253,17 +253,44 @@ If you want to restrict navigation between your slides (i.e. the user has to tog
...
+     * Toast.makeText(getContext(), "Please click on the checkbox to continue", Toast.LENGTH_SHORT).show();
+     */
+        // When user attempts to continue without fulfilling the policy this function is runned.
     }
 }
 ```
+Finally add the slide to your intro by using the following method :
+```
+addSlide(getSupportFragmentManager().getFragmentFactory().instantiate(getClassLoader(),MySlide.class.getName()));
+```
```
Comment (cortinico) [inline README.md:292]: Let's take only this part and let's change it with:

```
addSlide(MySlide.newInstance());
```

File: README.md [hunk 2]
```diff
@@ -253,17 +253,44 @@ If you want to restrict navigation between your slides (i.e. the user has to tog
...
+	// Once return value is true the user will be allowed to the next slide and vice versa.
     }
 
     @Override
     public void onUserIllegallyRequestedNextPage() {
-        // User illegally requested next slide
+    /* Example part 4 (Display a toast when user doesn't click on checkbox and attempts to move to next slide) :
+     * Toast.makeText(getContext(), "Please click on the checkbox to continue", Toast.LENGTH_SHORT).show();
+     */
+        // When user attempts to continue without fulfilling the policy this function is runned.
```
Comment (cortinico) [inline README.md:285]: This line should suffice, please remove the other.

File: README.md [hunk 3]
```diff
@@ -253,17 +253,44 @@ If you want to restrict navigation between your slides (i.e. the user has to tog
...
+	
+        return view;
+    }
     @Override
     public boolean isPolicyRespected() {
-        return // If user should be allowed to leave this slide
+    /* Example part 3 (Returns true if the user checked the checkbox and false if he didn't):
+     * return checkBox.isChecked();
+     */
+        return // You should return true if policy is respected or false if it isn't.
```
Comment (cortinico) [inline README.md:276]: Please keep only this line and remove the others.

File: README.md [hunk 4]
```diff
@@ -253,17 +253,44 @@ If you want to restrict navigation between your slides (i.e. the user has to tog
...
+ * private CheckBox checkBox;
+ */
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.myslide_layout, container, false);
+	
+	/* Usage Example part 2 (Assign the CheckBox view to the variable) :
+	 * checkBox = view.findViewById(R.id.checkboxid);
+	 */
```
Comment (cortinico) [inline README.md:266]: This whole part can probably go.

File: README.md (hunk 1 above)
Comment (Supermiro19) [reply README.md:292]: Well if the new instance method worked I would have used it but here is the result.
https://imgur.com/a/nUIhlry

The method I used is easier and simpler for everyone.

File: README.md (hunk 4 above)
Comment (Supermiro19) [reply README.md:266]: I meant to make changes to the documentation understandable even by newbie developers. The whole pull request was meant to just give the user a full step by step guide.

File: README.md (hunk 3 above)
Comment (Supermiro19) [reply README.md:276]: This line is meant to complete the others. To make a full step by step documentation.

File: README.md (hunk 2 above)
Comment (Supermiro19) [reply README.md:285]: As I mentioned in my other comments on the pull request I suggest that we leave it to give the users of the library a step by step documentation with examples.

**Comment by cortinico** [comment]
More context on this is on #759

**Pull request closed without merging**